routing/router.go

```diff
@@ -178,7 +178,7 @@ type MissionController interface {
 	// input for future probability estimates. It returns a bool indicating
 	// whether this error is a final error and no further payment attempts
 	// need to be made.
-	ReportPaymentFail(rt *route.Route,
+	ReportPaymentFail(paymentID uint64, rt *route.Route,
```

Comment:
what's the purpose of this change?

Reply:
To make reporting to mission control idempotent. Without this id, it wouldn't be able to recognize when it is offered the same result again after restart.